rocAL/source/decoders/image/rocjpeg_decoder.cpp

```diff
@@ -0,0 +1,386 @@
+                                               size_t max_decoded_width, size_t max_decoded_height,
+                                               std::vector<size_t> original_image_width, std::vector<size_t> original_image_height,
+                                               std::vector<size_t> &actual_decoded_width, std::vector<size_t> &actual_decoded_height) {
+
+
+    if (_resize_batch) {
+        // Allocate memory for the itermediate decoded output
+        _rocjpeg_image_buff_size *= _num_channels;
+        if (!_rocjpeg_image_buff) {
+            CHECK_HIP(hipMalloc((void **)&_rocjpeg_image_buff, _rocjpeg_image_buff_size));
+            _prev_image_buff_size = _rocjpeg_image_buff_size;
+        } else if (_rocjpeg_image_buff_size > _prev_image_buff_size) {  // Reallocate if the intermediate output exceeds the allocated memory
+            CHECK_HIP(hipFree((void *)_rocjpeg_image_buff));
+            CHECK_HIP(hipMalloc((void **)&_rocjpeg_image_buff, _rocjpeg_image_buff_size));
+            _prev_image_buff_size = _rocjpeg_image_buff_size;
+        }
+
+        uint8_t *img_buff = reinterpret_cast<uint8_t*>(_rocjpeg_image_buff);
+        size_t src_offset = 0;
+
+        // Update RocJpegImage with the pointer
+        for (unsigned i = 0; i < _batch_size; i++) {
+                _output_images[i].channel[0] = static_cast<uint8_t *>(img_buff);    // For RGB
+                _src_img_offset[i] = src_offset;
+                unsigned pitch_width = (original_image_width[i] + 8) &~ 7;
+                unsigned pitch_height = (original_image_height[i] + 8) &~ 7;
+                src_offset += (pitch_width * pitch_height * _num_channels);
+                img_buff += (pitch_width * pitch_height * _num_channels);
+                _src_hstride[i] = pitch_width * _num_channels;
+        }
+
+        // Copy width and height args to HIP memory
+        CHECK_HIP(hipMemcpyHtoD((void *)_dev_src_width, original_image_width.data(), _batch_size * sizeof(size_t)));
+        CHECK_HIP(hipMemcpyHtoD((void *)_dev_src_height, original_image_height.data(), _batch_size * sizeof(size_t)));
+        CHECK_HIP(hipMemcpyHtoD((void *)_dev_dst_width, actual_decoded_width.data(), _batch_size * sizeof(size_t)));
+        CHECK_HIP(hipMemcpyHtoD((void *)_dev_dst_height, actual_decoded_height.data(), _batch_size * sizeof(size_t)));
+        CHECK_HIP(hipMemcpyHtoD((void *)_dev_src_hstride, _src_hstride.data(), _batch_size * sizeof(size_t)));
+        CHECK_HIP(hipMemcpyHtoD((void *)_dev_src_img_offset, _src_img_offset.data(), _batch_size * sizeof(size_t)));
+    } else {
+        for (unsigned i = 0; i < _batch_size; i++) {
+            _output_images[i].channel[0] = static_cast<uint8_t *>(output_buffer[i]);    // For RGB
+        }
+    }
+
+    CHECK_ROCJPEG(rocJpegDecodeBatched(_rocjpeg_handle, _rocjpeg_streams.data(), _batch_size, &_decode_params, _output_images.data()));
```

Comment:
As discussed earlier the `_decode_params ` should be changed to a vector so please change the `&_decode_params` to `_decode_params.data()`

Reply:
Done